core/src/config/mod.rs

```diff
@@ -0,0 +1,157 @@
+pub mod payload;
+pub mod yaml;
+
+use serde::{Deserialize, Serialize};
+use smol::stream::StreamExt;
+use std::{
+    error::Error,
+    fmt::Debug,
+    fs::{self, read_dir, OpenOptions},
+    path::Path,
+};
+use tracing::{debug, info_span};
+
+use crate::{
+    ordering::{construct_groups, resolve_before, sort},
+    validate,
+};
+use tracing::{error, instrument};
+
+use self::{payload::Payload, yaml::TaskConfigYaml};
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub enum Respawn {
+    /// Never retry this task (default)
+    No,
+    /// Restart this task up to N times
+    ///
+    /// N = 0, restart this task an unlimited number of times
+    // TODO: Does manual restart affect the counter, if so: how
+    Retry(usize),
+}
+
+impl Default for Respawn {
+    fn default() -> Self {
+        Self::No
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct TaskConfig {
+    pub name: String,
+    // #[serde(default)]
+    pub payload: Payload,
+    // #[serde(default)]
+    pub with: Vec<String>,
+    // #[serde(default)]
+    pub after: Vec<String>,
+    // #[serde(default)]
+    pub respawn: Respawn,
+    pub group: Option<String>,
+}
+
+impl TaskConfig {
+    pub fn new(name: String) -> Self {
+        Self {
+            name,
+            ..Default::default()
+        }
+    }
+
+    pub fn after(&mut self, name: &str) -> &mut Self {
+        self.after.push(name.to_owned());
+        self
+    }
+}
+
+pub fn read_config(builtin: Vec<TaskConfigYaml>) -> Vec<TaskConfig> {
+    let configs = if cfg!(debug_assertions) {
+        "test"
+    } else {
+        "/etc/alfad"
```

Comment:
Probably this is a good idea to have these as constants somewhere, so `Path::new(somewhere.DIR_CFG)` for example.

Reply:
@Ichmed BTW, what's wrong with:

```rust
let configs = Path::new(if cfg!(debug_assertions) { somewhere.DIR_TEST } else { somewhere.DIR_CFG });
```

...?